### DISCUSSION TITLE
[archived] "HTTP Template" docs discussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#12718] [General]
# en/latest/http-template/

https://argo-workflows.readthedocs.io/en/latest/http-template/

**Comment by dmuley17** [comment]
Hello Team, I am new to argo and trying to use the HTTP Template 
I have used the role https://github.com/argoproj/argo-workflows/blob/main/manifests/quick-start/base/agent-role.yaml
for agent but still facing the issue 

` MountVolume.SetUp failed for volume "kube-api-access-96r26" : secret "argo.service-account-token" not found`

any idea what could be wrong ?

**Reply by dmuley17** [reply]
FYI - I am not using token for authentication or authorization.

**Reply by agilgur5** [reply]
If you're using the Quick Start manifests, the `default` SA has everything set up. In this case, you're using the `argo` SA and it's token can't be found; `default` [has one](https://github.com/argoproj/argo-workflows/blob/7ba20fea08e40a5792a1d25011ee8c12889960c5/manifests/quick-start/base/default.service-account-token-secret.yaml#L4) appropriately named. Since k8s 1.24, Secrets must be [manually created for SAs](https://argo-workflows.readthedocs.io/en/latest/manually-create-secrets/)

**Reply by dmuley17** [reply]
thank you for your response I tried with both 
` argo submit -n argo http-hello-world.yaml --serviceaccount default --watch` 

but same error 

`MountVolume.SetUp failed for volume "kube-api-access-nwl9w" : secret "default.service-account-token" not found`

**Reply by agilgur5** [reply]
Did you create the token with the Quick Start [manifest](https://github.com/argoproj/argo-workflows/blob/7ba20fea08e40a5792a1d25011ee8c12889960c5/manifests/quick-start/base/default.service-account-token-secret.yaml#L4)? It sounds like you might have not installed using the Quick Start manifests?

**Reply by dmuley17** [reply]
I just scrapped and recreated argo server now its working. I followed the quick start though. Thank you for response.

**Comment by mikewy-quantpi** [comment]
By following [this guide](https://argo-workflows.readthedocs.io/en/latest/access-token/) to create a secret for service account argo-agent will solve the issue.

**Comment by shenshanyoumu** [comment]
always pending status, my argo version is 3.5.5

**Reply by agilgur5** [reply]
This isn't an issue, there are newer patches, and this would need a lot more details to root cause.
Try with 3.5.6, and if you have an error due to Argo, fill out a reproducible bug report.
